### PR TITLE
Feature/#10 Create roll dice button

### PR DESCRIPTION
**Pull request by FFreestyler**

### CODE REVIEW

**Comment by FFreestyler** [comment]
[Create roll dice button](https://trello.com/c/NkEYjfiF/10-create-roll-dice-button)

File: src/components/RollDiceButton.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+}));
+
+const StyledMenu = styled(Menu)(({ theme }) => ({
+  '& .MuiPaper-root': {
+    backgroundColor: 'rgba(0, 0, 0, 0)',
+    boxShadow: 'none',
+  },
+}));
+
+function RollDiceButton() {
```
Comment (MangriMen) [inline src/components/RollDiceButton.tsx]: Need to be arrow function component with FC type (typescript).

File: src/components/RollDiceButton.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+              </StyledMenuItem>
+            </Tooltip>
+          ))}
+        </StyledMenu>
+      </FormControl>
+    </ParBox>
+  );
+}
+
+export default RollDiceButton;
```
Comment (MangriMen) [inline src/components/RollDiceButton.tsx]: Need to be moved in folder, maybe components/charlist/?

File: src/components/RollDiceButton.tsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+          borderRadius: '50%',
+        }}
+      >
+        {open ? (
+          <CloseIcon fontSize="large" sx={{ color: grey[50] }} />
+        ) : (
+          <Box component="img" src={D20Icon} />
+        )}
+      </Button>
+      <FormControl>
```
Comment (MangriMen) [inline src/components/RollDiceButton.tsx]: What is the FormControl for?

File: src/components/RollDiceButton.tsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        <StyledMenu
```
Comment (MangriMen) [inline src/components/RollDiceButton.tsx]: Animation is default for menu, we need beautiful version like in old html charlist i think.

File: src/components/RollDiceButton.tsx [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+import D8Icon from 'assets/images/dice/D8.svg';
+import D10Icon from 'assets/images/dice/D10.svg';
+import D12Icon from 'assets/images/dice/D12.svg';
+import D20Icon from 'assets/images/dice/D20.svg';
+import D100Icon from 'assets/images/dice/D100.svg';
+import React from 'react';
+
+import ParBox from './styled/ParBox';
+
+interface DiceMenuItem {
```
Comment (MangriMen) [inline src/components/RollDiceButton.tsx]: Once the RollDiceButton has been moved to the components/charlist/ folder, create interfaces.ts and move it there.

**Review by MangriMen — CHANGES_REQUESTED** [review]

File: src/components/charlist/MainPage.tsx [hunk 6]
```diff
@@ -3,6 +3,7 @@ import { FC } from 'react';
 
 import ParBox from '../styled/ParBox';
```
Comment (StepanovRoma) [inline src/components/charlist/MainPage.tsx]: Fix the path to components/styled/

File: src/components/charlist/MainPage.tsx [hunk 7]
```diff
@@ -3,6 +3,7 @@ import { FC } from 'react';
...
 import ParContainer from '../styled/ParContainer';
```
Comment (StepanovRoma) [inline src/components/charlist/MainPage.tsx]: It's the same here too

File: src/components/charlist/RollDiceButton.tsx [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const dice: Array<DiceMenu> = [
+    {
+      name: 'D20',
+      iconName: D20Icon,
+      onClick: () => {
```
Comment (StepanovRoma) [inline src/components/charlist/RollDiceButton.tsx]: You can write this onClick in the map and put all the data about the dices in another file

File: src/components/charlist/RollDiceButton.tsx [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+        }}
+      >
+        {dice.map((dices) => (
+          <Tooltip
+            disableInteractive
+            placement="left"
+            key={dices.name}
+            title={<Typography> {dices.name || ''} </Typography>}
+          >
+            <StyledMenuItem onClick={dices.onClick}>
```
Comment (StepanovRoma) [inline src/components/charlist/RollDiceButton.tsx]: Here just replace with this:
onClick={()=>handleClose()}

**Review by StepanovRoma — CHANGES_REQUESTED** [review]

File: src/components/charlist/RollDiceButton.tsx (hunk 9 above)
Comment (FFreestyler) [reply src/components/charlist/RollDiceButton.tsx]: No

**Review by StepanovRoma — APPROVED** [review]

File: src/components/charlist/RollDiceButton.tsx [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const dice: Array<DiceMenu> = [
+    {
+      name: '20',
```
Comment (IHiFiveI) [inline src/components/charlist/RollDiceButton.tsx]: As discussed earlier, may be rename this to "edges" and change value type to string? Like `{ name: '20' }` to `{ edges: 20 }`

File: src/components/charlist/RollDiceButton.tsx [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+          width: '100%',
+          justifyContent: 'center',
+        }}
+      >
+        {dice.map((dices) => (
+          <Tooltip
+            disableInteractive
+            placement="left"
+            key={dices.name}
+            title={<Typography> {'D' + dices.name || ''} </Typography>}
```
Comment (IHiFiveI) [inline src/components/charlist/RollDiceButton.tsx]: Rewrite concat using backticks

File: src/components/charlist/RollDiceButton.tsx [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+        anchorEl={anchorEl}
+        open={open}
+        onClose={handleClose}
+        sx={{
+          display: 'flex',
+          width: '100%',
+          justifyContent: 'center',
+        }}
+      >
+        {dice.map((dices) => (
```
Comment (IHiFiveI) [inline src/components/charlist/RollDiceButton.tsx]: I think naming is a bit messed up there

**Review by IHiFiveI — APPROVED** [review]

**Review by MangriMen — APPROVED** [review]

**Pull request merged**